### ISSUE TITLE
trouble with dedicating multiple GPU speech_to_text_bpe

### ISSUE THREAD

**Issue by sarasesar** [NVIDIA/NeMo#3300]
Hi, I've trouble with dedicating multiple GPU for asr/speech_to_text_bpe.py.
In config file of conformer, I tried gpu=-1 and also gpu=2 but when I run the code, it just use the first GPU and it doesn't use the second one.
Could you please help me to solve it?
As you see when I run the code, it use just the first GPU and just use 11Mb of the second one.

![Screenshot from 2021-12-08 16-58-13](https://user-images.githubusercontent.com/95332871/145216745-c1413158-3836-489b-a3c6-58f32df45470.png)

**Comment by shahin-trunk** [comment]
Should we use trainer.accelerator="ddp" or trainer.strategy="ddp" ?

**Comment by titu1994** [comment]
Hmm for the moment, accelerator. We will be updating to the new syntax in an upcoming release.

**Comment by sarasesar** [comment]
Thank you a galaxy! I tried the branch that you said and it worked.
Now it uses both GPUs but I have NCCL trouble.
I've solved it by this code:
os.environ['MASTER_ADDR'] = 'localhost'
os.environ['MASTER_PORT'] = '5678'
dist.init_process_group(backend='nccl', init_method='env://', rank = 0, world_size = 1)

I added the above code in speech_to_text_bpe.py and the NCCL problem solved but when I add this to my code, it just use the first GPU. But when I don't use this in my code, I have NCCL problem.

how can I solve the NCCL problem?

Full context of the problem:
RuntimeError: NCCL error in: ../torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:957, unhandled system error, NCCL version 21.0.3
ncclSystemError: System call (socket, malloc, munmap, etc) failed.
